external-builds/pytorch/build_pytorch_windows.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+# Windows equivalent to the Linux build_pytorch_torch.sh.
+# TODO: merge into `ptbuild.py develop` or `ptbuild.py bdist_wheel` somehow?
+#
+# Usage (choose your gfx target to match -DTHEROCK_AMDGPU_FAMILIES from the build):
+#   bash build_pytorch_windows.sh gfx1100
+#   bash build_pytorch_windows.sh gfx1151
+
+set -eo pipefail
+
+if [ -n "$1" ]; then
+    export PYTORCH_ROCM_ARCH=$1
+    echo "Setting PYTORCH_ROCM_ARCH to $PYTORCH_ROCM_ARCH"
+elif [ -z "$PYTORCH_ROCM_ARCH" ]; then
+    echo "PYTORCH_ROCM_ARCH must be set as an env var or passed as a script arg"
+    exit 1
+fi
+
+SCRIPT_DIR="$(cd $(dirname $0) && pwd)"
+
+if [ ! -n "${ROCM_HOME}" ]; then
+    export ROCM_HOME=$(realpath $SCRIPT_DIR/../../build/dist/rocm)
+    echo "ROCM_HOME: $ROCM_HOME"
+fi
+if [ -d ${ROCM_HOME} ]; then
+    export PATH=${ROCM_HOME}/bin:$PATH
+else
+    echo "Could not find ROCM_HOME: $ROCM_HOME"
+    exit 1
+fi
+
+BUILD_DIR_ROOT=${ROCM_HOME?}/../..
+
+# Environment variables recommended here:
+# https://github.com/ROCm/TheRock/discussions/409#discussioncomment-13032345
+export USE_ROCM=ON
+export USE_KINETO=0
+export BUILD_TEST=0
+export USE_FLASH_ATTENTION=0
+export USE_MEM_EFF_ATTENTION=0
+export CMAKE_PREFIX_PATH="${ROCM_HOME?}"
+# TODO(#410): Fix HIP_CLANG_PATH setting so hipcc finds the tools on its own
+export HIP_CLANG_PATH="${ROCM_HOME?}/lib/llvm/bin"
+export CC=${ROCM_HOME?}/lib/llvm/bin/clang-cl
+export CXX=${ROCM_HOME?}/lib/llvm/bin/clang-cl
+export DISTUTILS_USE_SDK=1
+
+# Alternate paths relative to this script: 'src/', 'src/pytorch/'.
+PYTORCH_SRC_DIR="${SCRIPT_DIR?}/pytorch"
+cd ${PYTORCH_SRC_DIR}
```

Comment:
Note that this path is populated by https://github.com/ROCm/TheRock/blob/main/external-builds/pytorch/pytorch_torch_repo.py, _not_ `src/` that `ptbuild.py` currently uses.

Once https://github.com/ROCm/TheRock/pull/620 or a similar change goes in, I'll update this to the single path we've decided on (I'm proposing `src/pytorch` in that PR).